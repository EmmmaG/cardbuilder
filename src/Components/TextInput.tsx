import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Select from '@material-ui/core/Select'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'

const styles = (theme: Theme) => ({
	inputContainer: {
		textAlign: 'left' as 'left',
		marginTop: '5px',
		marginBottom: '5px',
		marginLeft: '10px',
		marginRight: '10px',
		display: 'inline-block',
	},
	textField: {
		height: '35px',
		backgroundColor: 'white',
		width: '195px',
	},
	inputRoot: {
		height: '35px',
		fontFamily: '"Big Caslon", "Times New Roman", "serif"',
	},
	title: {
		color: theme.palette.grey[700],
		marginBottom: '2px',
	},
})

export interface Select {
	value: string
	text: string
}

export interface Props extends WithStyles<typeof styles> {
	title: string
	onChange: (selectedValue: string) => void
}

export interface State {
    value: string
}

export class TextInput extends React.Component<Props, State> {
    public state = { value: ''}

	private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const value = event.target.value

        this.setState({ value }, () => this.props.onChange(value))
	}

	public render() {
        const { classes, title } = this.props
        const { value } = this.state

		return (
			<div className={classes.inputContainer}>
				<Typography className={classes.title}>{title}</Typography>
				<TextField
					id={title}
					className={classes.textField}
					value={value}
					onChange={this.handleChange}
					variant='outlined'
					InputProps={{ classes: { root: classes.inputRoot } }}
				/>
			</div>
		)
	}
}

export default withStyles(styles)(TextInput)
