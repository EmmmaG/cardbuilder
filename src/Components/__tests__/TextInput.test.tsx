import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'
import TextField from '@material-ui/core/TextField'

import TextInput, { Props } from '../TextInput'

const props: Props = {
    title: 'Details',
    onChange: jest.fn(),
	classes: idObj,
}

describe('TextInput', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<TextInput {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
    })

    it('updates with new value', () => {
        const wrapper = shallow(<TextInput {...props} />).dive()
        const textField = wrapper.find(TextField)
        const newValue = 'new'
        const event = { target: {value: newValue}} as any

        textField.prop('onChange')!(event)

        expect(wrapper.state('value')).toEqual(newValue)
        expect(props.onChange).toHaveBeenCalledWith(newValue)
    })
})
