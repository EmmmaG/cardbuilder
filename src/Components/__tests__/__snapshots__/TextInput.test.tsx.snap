// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<div
  className="TextInput-inputContainer-1"
>
  <WithStyles(Typography)
    className="TextInput-title-4"
  >
    Details
  </WithStyles(Typography)>
  <TextField
    InputProps={
      Object {
        "classes": Object {
          "root": "TextInput-inputRoot-3",
        },
      }
    }
    className="TextInput-textField-2"
    id="Details"
    onChange={[Function]}
    required={false}
    select={false}
    value=""
    variant="outlined"
  />
</div>
`;
